include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -1004,13 +1023,56 @@ bool adjacent_stringize = false;
                     impl::replace_ellipsis(expanded_args, i, expanded, pos);
                 }
                 else
+
+#if BOOST_WAVE_SUPPORT_VA_OPT != 0
+                if (is_va_opt) {
+                    position_type const &pos = (*cit).get_position();
+
+                    BOOST_ASSERT(boost::wave::need_va_opt(ctx.get_language()));
+
+                    // ensure all variadic arguments to be expanded
+                    for (typename vector<ContainerT>::size_type arg = i;
+                         arg < expanded_args.size(); ++arg)
+                    {
+                        expand_argument(arg, arguments, expanded_args,
+                            expand_operator_defined, has_expanded_args);
+                    }
+
+                    // locate the __VA_OPT__ arguments
+                    typename macro_definition_type::const_definition_iterator_t cstart = cit;
+                    if (!impl::find_va_opt_params(cit, cend)) {
+                        BOOST_WAVE_THROW_CTX(ctx, preprocess_exception,
+                            improperly_terminated_macro, "missing ')' in __VA_OPT__",
+                            pos);
+                    }
+                    // cstart still points to __VA_OPT__; cit now points to the last rparen
+
+                    if ((i == arguments.size()) ||             // no variadic argument
+                        impl::is_blank_only(arguments[i])) {   // no visible tokens
+                        // no args; insert placemarker
+                        *std::back_inserter(expanded) =
+                            typename ContainerT::value_type(T_PLACEMARKER, "\xA7", pos);
+                    } else {
+                        ++cstart; ++cstart;
+                        // [cstart, cit) is now the args to va_opt
+                        // recursively process them
+                        expand_replacement_list(cstart, cit, arguments,
+                                                expand_operator_defined, expanded);
+                    }
```

Comment:
I'm not sure if recursively expanding the replacement list is the correct solution here. I think calling `impl::replace_ellipsis` (like for `__VA_ARGS__` above) in connection with the rescanning that is done afterwards  should be sufficient. Doing recursive expanding of the replacement list might actually expand too much.

Reply:
I'll give that a try. Do you think this line from the proposal changes anything?

> Otherwise, the replacement consists of the results of the expansion of content as the replacement list of the current function-like macro before rescanning and further replacement

Reply:
Hmmm, I think you're right. `__VA_OPT__` is to be treated as a function like macro... Please disregard this comment.

Reply:
OK so I replaced the call to `expand_replacement_list` with this simple copy (which seems to be what `replace_ellipsis` does):
```c++
std::copy(cstart, cit, std::inserter(expanded, expanded.end()));
```
and I get some unit test failures, all of which seem to be that `__VA_ARGS__` is not replaced in the output.